### PR TITLE
Update dskit to 7b512eb.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This is a small update to dskit that includes https://github.com/grafana/dskit/pull/406 which will allow us to intercept gRPC request in call clients.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Review by MichelHollands — APPROVED** [review]
LGTM

File: pkg/analytics/seed_test.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func createMemberlist(t *testing.T, port, memberID int) *memberlist.KV {
 	var cfg memberlist.KVConfig
 	flagext.DefaultValues(&cfg)
 	cfg.TCPTransport = memberlist.TCPTransportConfig{
-		BindAddrs: []string{"localhost"},
+		BindAddrs: []string{"0.0.0.0"},
```
Comment (dannykopping) [inline pkg/analytics/seed_test.go:39]: What's the reason for this change?

**Review by dannykopping — APPROVED** [review]
LGTM, but curious why the test needs to bind on `0.0.0.0`

File: pkg/analytics/seed_test.go (hunk 1 above)
Comment (jeschkies) [reply pkg/analytics/seed_test.go:39]: See https://github.com/grafana/dskit/pull/396. It's a security fix.

Comment (dannykopping) [reply pkg/analytics/seed_test.go:39]: Huh? The security fix was to correctly bind to localhost, precisely to _not_ bind on all interfaces inadvertently. In any case, this is a test so no worries, was just curious.
